python/lsst/validate/drp/validate.py

```diff
@@ -202,6 +209,22 @@ def runOneRepo(repo, dataIds=None, metrics=None, outputPrefix='', verbose=False,
     The filter name is added to this prefix.  If the filter name has spaces,
     there will be annoyance and sadness as those spaces will appear in the filenames.
     """
+    from lsst.daf.persistence import Butler
+    if instrument is None:
+        instrument = extract_instrument_from_repo(butler)
+    if dataset_repo_url is None:
+        dataset_repo_url = repo
+
+    def extract_instrument_from_repo(repo):
+        """Extract the last part of the mapper name from a Butler repo.
+        'lsst.obs.lsstSim.lsstSimMapper.LsstSimMapper' -> 'lsstSim'
+        'lsst.obs.cfht.megacamMapper.MegacamMapper' -> 'cfht'
+        'lsst.obs.decam.decamMapper.DecamMapper' -> 'decam'
+        'lsst.obs.hsc.hscMapper.HscMapper' -> 'hsc'
+        """
+        mapper_class = Butler.getMapperClass(repo)
+        instrument = mapper_class.split('.')[2]
+        return instrument
```

Comment:
For what it's worth, we have (not uniformly unfortunately) been using all upper case for instrument names.  Maybe you should `return instrument.upper()`.

Reply:
I'd like to keep with the simpler beahvior.
I'm not trying to replicate the SQuaSH behavior.
For use with SQuaSH, the caller can specify the instrument name.

Reply:
I didn't understand that 'instrument' was required now because the 'verify_metrics' are now only specified for defined instruments.  And thus 'instrument' must be something that matches to something specified in a metrics YAML file.  That's why it matters that it's upper-case.